### PR TITLE
Containerize

### PR DESCRIPTION
**Pull request by fosskers**
### TODO
- [x] Dockerfile
- [x] Makefile
- [x] Update README

### Motivation
We're updating these old demos to run in a containerized environment, so that they can be ran elsewhere easier.

**Note:** At least on my machine, I needed to include `--net=host` when running the image:

```
docker run --rm -p 8777:8777 --net=host -t chatta:latest
```

**Note:** Depends on #27

### CODE REVIEW

**Comment by pomadchin** [comment]
@fosskers you don't need to forward ports if you use `--net=host` flag, though I'm wondering what else it's necessary to forward to make it work without `--net=host`.

**Comment by fosskers** [comment]
I don't know. Without `--net=host` it just doesn't work.

**Comment by pomadchin** [comment]
Investigated: should be correctly setuped binding hostname in [server config](https://github.com/geotrellis/geotrellis-chatta-demo/blob/master/geotrellis/src/main/resources/application.conf#L5). 

A possible container launch would be smth like that: 

```bash
docker run -d -p 8777:8777 -h <hostname> chatta:latest
```

Where `<hostname>` is a host machine name, different from localhost (for example `ip-172-31-13-105`). It's akka http feature. As a workaround it is possible to use host machine network from the container (as you described, using `--net=host` flag):

```bash
docker run -d --net=host chatta:latest
```

Everything should be explained in docs.

**Comment by fosskers** [comment]
Yup, setting `-Dgeotrellis.hostname=chatta` in the Dockerfile, then running with `docker run --rm -p 8777:8777 -h chatta -t chatta:latest` works.

File: geotrellis/build.sbt [hunk 1]
```diff
@@ -30,18 +30,24 @@ libraryDependencies ++= Seq(
   "org.apache.hadoop"  % "hadoop-client" % "2.7.3"
 )
 
-test in assembly := {}
+dependencyOverrides += "com.google.guava" % "guava" % "16.0.1"
```
Comment (pomadchin) [inline geotrellis/build.sbt]: Unnecessary line.

Comment (fosskers) [inline geotrellis/build.sbt]: Ok I'll remove it and try everything.

**Pull request merged**